src/components/TransactionList.js

```diff
@@ -1,14 +1,16 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Transaction } from './Transaction';
 
 export const TransactionList = () => {
+    const [transactions] = useState(
+        JSON.parse(localStorage.getItem('transactions')) || []
+    );
+
     return (
         <>
             <h4>History</h4>
             <ul className="list">
-                <li className="minus">
-                    Cash <span>-$400</span>
-                    <button className="delete-btn">x</button>
-                </li>
+                {transactions.map(transaction => (<Transaction key={transaction.id} transaction={transaction} />))}
```

Comment:
This part shows all transactions from localStorage, but if a transaction is deleted, how do I call this again?

Reply:
This component should know somehow about the changes. If you would like to keep the current implementation, `transactions` should sit on the level above (see [Lifting State Up](https://reactjs.org/docs/lifting-state-up.html) for better understanding), so that any child components can use it and modify it. So, the `App` will take care of the `transactions` state and the `useEffect` hook will sync it with local storage.

Reply:
Thanks for the feedback! I'll fix this.